### ISSUE TITLE
StackOverflowError

### ISSUE THREAD

**Issue by yoobi** [jraska/modules-graph-assert#227]
I've recently started to add rules using the `moduleGraphAssert` but I'm getting this error, I'm not sure if this is a normal error when graph should fail.
```
Cause: java.lang.StackOverflowError
	at kotlin.TuplesKt.to(Tuples.kt:43)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:83)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:84)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:84)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:84)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:84)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:84)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:84)
	at com.jraska.module.graph.DependencyGraph.addConnections(DependencyGraph.kt:84)
```

My rules are the following:
```
moduleGraphAssert {
    maxHeight = 5
    allowed = [
        ":app -> :.*",
        ":features:.* -> :.*",
        ":features:feature-[a-z]*:data -> :features:feature-[a-z]*:domain",
        ":features:feature-[a-z]*:ui -> :features:feature-[a-z]*:domain",
    ]
    restricted = [
        ":.* -X> :app",
        ":features:feature-[a-z]*:domain -X> :features:feature-[a-z]*:ui",
        ":features:feature-[a-z]*:data -X> :features:feature-[a-z]*:ui",
    ]
    assertOnAnyBuild = true
}
```
In order to test I've purposely made `features:feature-a:domain` implement `features:feature-a:ui`
My architecture looks like this:
- app
- core
    - network
- features
    - feature-a
        - data
        - domain
        - ui
    - feature-b
        - data
        - domain
        - ui
    - feature-c
        - data
        - domain
        - ui

**Comment by jraska** [comment]
Hey, this happens when there are cyclic dependencies within your module graph - cycles are not allowed.

I think Gradle itself will fail as well even without the module-graph-assert plugin.
The plugin likely fails just even before now so I would remove the plugin, see where the cycle is and then potentially add the plugin back :)

**Comment by yoobi** [comment]
Hello, 

Thanks for your fast reply ! I didn't think about cyclic dependencies but it is indeed the case. 
I have the following dependencies in features `data --> domain <-- ui` and I've added an implementation of ui in domain.

Well I think you can close this issue haha, sorry :)

**Comment by Keith-Albright-Bose** [comment]
fyi gradle sync by itself doesn't fail, but of course trying to later use an import and some code will result in stack overflow.
So it'd be nice to catch that exception and msg something.

**Comment by jraska** [comment]
@Keith-Albright-Bose please try out the latest version. The issue should be solved by #264
